internal/daemon/daemon.go

```diff
@@ -124,7 +124,14 @@ func (d *Daemon) Init(ctx context.Context, listenPort string, stateDir string, s
 
 	close(d.ReadyChan)
 
-	return nil
+	for {
+		select {
+		case <-ctx.Done():
+			d.shutdownDoneCh <- d.stop()
```

Comment:
It seems a bit odd to depend on the send to the channel to return below here (although i appreciate we do need to send to the channel).

I would suggest we read do:

```go
err = d.stop()
d.shutdownDoneCh <- err
return err
```

for clarity and decouple the case where there is an err send to d.shutdownDoneCh from externally.